01.md

```diff
@@ -1,177 +1,235 @@
+This NIP defines the `kind:0` as **User Metadata**. The `.content` is set to a stringified JSON object `{name: <user's name, string>, about: <string>, picture: <url, string>}`. Since other attributes might exist, Clients SHOULD preserve any unsupported attribute when updating this event. 
 
-And also a convention for kind ranges that allow for easier experimentation and flexibility of relay implementation:
+## Tags
 
-- for kind `n` such that `1000 <= n < 10000`, events are **regular**, which means they're all expected to be stored by relays.
-- for kind `n` such that `10000 <= n < 20000 || n == 0 || n == 3`, events are **replaceable**, which means that, for each combination of `pubkey` and `kind`, only the latest event MUST be stored by relays, older versions MAY be discarded.
-- for kind `n` such that `20000 <= n < 30000`, events are **ephemeral**, which means they are not expected to be stored by relays.
-- for kind `n` such that `30000 <= n < 40000`, events are **parameterized replaceable**, which means that, for each combination of `pubkey`, `kind` and the `d` tag's first value, only the latest event MUST be stored by relays, older versions MAY be discarded.
+Each tag is an array of strings of arbitrary size. Their meaning is determined by the event `.kind`. Tags with the same name might have entirely different meanings in different kinds. 
 
-In case of replaceable events with the same timestamp, the event with the lowest id (first in lexical order) should be retained, and the other discarded.
+The first element of the tag array is referred to as the tag _name_ and the second as the tag _value_. All elements after the second do not have a conventional name.
 
-When answering to `REQ` messages for replaceable events such as `{"kinds":[0],"authors":[<hex-key>]}`, even if the relay has more than one version stored, it SHOULD return just the latest one.
+All single-letter (only English alphabet letters: a-z, A-Z) tag names are indexed by relays for faster queries. 
 
-These are just conventions and relay implementations may differ.
+This NIP defines the format of 3 standard tags: `e`, `p`, and `a`. Tags `e`, `p` can be used to reference events and pubkeys respectively, and tag `a` references the latest version of a replaceable event. Clients MAY `a`- and `e`-tag parameterized replaceables simultaneously.
 
-## Communication between clients and relays
+| Name | Value                                                      | Other Params            |
+| ---- | ---------------------------------------------------------- | ----------------------- |
+| `e`  | `<32-byte lowercase hex of an event id>`                   | `<relay URL, optional>` |
+| `p`  | `<32-byte lowercase hex of a pubkey>`                      | `<relay URL, optional>` |
+| `a`  | `<kind>:<32-byte lowercase hex of a pubkey>:`              | `<relay URL, optional>` |
+| `a`  | `<kind>:<32-byte lowercase hex of a pubkey>:<d-tag value>` | `<relay URL, optional>` |
```

Comment:
Where is now #d ?

Reply:
Let me explain why I have difficulties with the "d" tag.
I looked at the README and saw: "Aha, there's an official #d. Let me have a look at what this means."
I then go to NIP-01 where it's supposed to be defined. I ctrl-f through the page on GitHub and don't find it.

After your PR here, when I ctrl-f, I'll still be confused because I'm lazy and I'll find two things:
* A statement saying 
  > This NIP defines the format of 3 standard tags: `e`, `p`, and `a`.
  
  Where the heck is now #d?
* A table with `e`, `p` and `a`
  Why the heck is it missing #d?